TheThingsNetwork/lorawan-stack
# Allow resending the contact info validation email in Console

#### Summary

<img width="904" alt="image" src="https://github.com/TheThingsNetwork/lorawan-stack/assets/5710611/92dd3c00-9dab-4024-9860-523f90d71486">

Closes #6498 

#### Changes



- Add store logic for `requestValidation`
- Add a button to the error screen of restricted users to enable resending the validation email
- Some related cleanup of authentication related errors

#### Testing

1. Configure the stack to require validated contact info `--is.user-registration.contact-info-validation.required`
2. Create a new user
3. Log into the console
4. Observe the new error page which includes a "Resend validation email" button
5. Click the button
6. It will likely say that an email has already been sent recently (as part of the registration)
7. Wait until the validation token expires (or delete it manually in the DB)
8. Click the button again
9. A success message appears and the email is sent again

#### Notes for Reviewers

I don't really like that you can only resend the email when the old token is not pending. If for whatever reason the first email got lost, there's no way for the user to trigger another email other than waiting until the token expires. cc @nicholaspcr 

cc @KrishnaIyer 

#### Checklist
